infrastructure/modules/infra/providers.tf

```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source                = "hashicorp/azurerm"
+      version               = "4.34.0"
```

Comment:
Can we remove the version pinning? Hopefully the root module provider version is used